app/code/core/Mage/Core/Model/Resource/Db/Abstract.php

```diff
@@ -656,10 +657,10 @@ protected function _checkUnique(Mage_Core_Model_Abstract $object)
 
                 if (is_array($unique['field'])) {
                     foreach ($unique['field'] as $field) {
-                        $select->where($field . '=?', trim($data->getData($field)));
+                        $select->where($field . '=?', trim((string)$data->getData($field)));
```

Comment:
```suggestion
                        if ($data->getData($field) !== null) {
                            $select->where($field . '=?', trim($data->getData($field)));
                        }
```

Reply:
It's probably better, I will try.

Reply:
No this does not work.

Reply:
I  tested this by saving an attribute set with an existing name, it works just fine.

> Attribute set with the "Default" name already exists.

What error do you get?